r2/metrics.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```

Comment:
nit: we tend to keep the metrics structs in the same file as the one the metrics struct is defined for. If not, perhaps rename this file to `r2/service_metrics.go`?